python-test-samples/step-functions-local-lambda/README.md

```diff
@@ -0,0 +1,259 @@
+export AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+export AWS_REGION='us-east-1'
+python3 -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+pip install -r requirements.txt
+```
+
+#### Run the unit tests
+
+``` shell
+step-functions-local-helloworld/tests$
+python3 -m pytest -s unit/src/test_step_functions_local.py
+```
+
+Expected output:
+``` shell
+step-functions-local-helloworld/tests$ python3 -m pytest -s unit/src/test_step_functions_local.py
+========================================================== test session starts ==========================================================
+platform linux -- Python 3.10.12, pytest-8.4.1, pluggy-1.6.0
+rootdir: /home/ubuntu/environment/serverless-test-samples-approach-2-pytest-distributed_Try3/step-functions-local-helloworld--OK/tests
+collected 1 item                                                                                                                                                                                                                                         
+
+unit/src/test_step_functions_local.py Pulling image testcontainers/ryuk:0.8.1
+Container started: ecdd9649b97d
+Waiting for container <Container: ecdd9649b97d> with image testcontainers/ryuk:0.8.1 to be ready ...
+Pulling image amazon/aws-stepfunctions-local
+Container started: d1bd33969e6b
+Waiting for container <Container: d1bd33969e6b> with image amazon/aws-stepfunctions-local to be ready ...
+.
+
+========================================================== 1 passed in 10.57s ==========================================================
+(
+```
+
+#### Clean up section
+
+> clean pyenv environment
+
+```sh
+step-functions-local-helloworld/tests$
+deactivate
+rm -rf venv/
+```
+
+> unsetting variables
+
+```sh
+unset AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
+unset AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+unset AWS_REGION='us-east-1'
+```
+
+> cleanning docker
+
+```sh
+docker ps --filter "ancestor=amazon/aws-stepfunctions-local" -q | xargs -r docker kill
+docker rmi amazon/aws-stepfunctions-local
+```
+
+#### Debug - PyTest Debugging
+
+For more detailed debugging in pytest:
+
+```sh
+# Run with verbose output
+python -m pytest -s -v unit/src/test_step_functions_local.py
+
+# Run with debug logging
+python -m pytest -s -v unit/src/test_step_functions_local.py --log-cli-level=DEBUG
+```
+
+---
+
+### Fast local development for Step Functions
+
+#### AWS CLI Commands for Manual Verification
+
+If you need to manually verify the state machine or execution details, you can use these commands:
+
+#### Configure environment variables:
+
+```sh
+step-functions-local-helloworld/tests$
+export AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
+export AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+export AWS_REGION='us-east-1'
+```
+
+#### Set up state machine machine manually
+
+```sh
+# Runing step functions docker image in the background
+step-functions-local-helloworld/tests$
+docker run -d --network host            \
```

Comment:
Same issue as `step-functions-local-helloworld`, Mac and Windows don't grant host network access in this mode (according to my research). I got a "Could not connect to the endpoint URL: http://localhost:8083" error.

Dropping the switch `--network host` worked for me.